CMakeLists.txt

```diff
@@ -237,6 +237,8 @@ set (HDF5_JAVA_LOGGING_NOP_JAR     ${HDF5_SOURCE_DIR}/java/lib/ext/slf4j-nop-1.7
 set (HDF5_JAVA_LOGGING_SIMPLE_JAR  ${HDF5_SOURCE_DIR}/java/lib/ext/slf4j-simple-1.7.33.jar)
 set (HDF5_DOXYGEN_DIR              ${HDF5_SOURCE_DIR}/doxygen)
 
+set (HDF5_SRC_INCLUDE_DIRS ${HDF5_SRC_DIR})
```

Comment:
Since the subfiling VFD is composed of several files, I tried to contain the feature to its own subdirectory under `src/`. CMake then needed to be adjusted so that everywhere ${HDF5_SRC_DIR} was added to a target's include directories, the subfiling subdirectory would be as well. This is so that the public H5FDsubfiling.h and H5FDioc.h headers are available for anything that includes hdf5.h.